src/super_gradients/training/utils/ema.py

```diff
@@ -89,3 +89,88 @@ def update_attr(self, model):
+    """ Model Exponential Moving Average from https://github.com/rwightman/pytorch-image-models
+    Keep a moving average of everything in the model state_dict (parameters and buffers).
+    This is intended to allow functionality like
+    https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage
+    A smoothed version of the weights is necessary for some training schemes to perform well.
+    This class is sensitive where it is initialized in the sequence of model init,
+    GPU assignment and distributed training wrappers.
+    """
+
+    def __init__(self, kd_model: KDModule, decay: float = 0.9999, beta: float = 15, exp_activation: bool = True):
+        """
+        Init the EMA
+        :param kd_model: Union[SgModule, nn.Module], the training model to construct the EMA model by
+                    IMPORTANT: WHEN THE APPLICATION OF EMA ONLY ON A SUBSET OF ATTRIBUTES IS DESIRED, WRAP THE NN.MODULE
+                    AS SgModule AND OVERWRITE get_include_attributes() AND get_exclude_attributes() AS DESIRED (SEE
+                    YoLoV5Base IMPLEMENTATION IN super_gradients.trainer.models.yolov5.py AS AN EXAMPLE).
+        :param decay: the maximum decay value. as the training process advances, the decay will climb towards this value
+                      until the EMA_t+1 = EMA_t * decay + TRAINING_MODEL * (1- decay)
+        :param beta: the exponent coefficient. The higher the beta, the sooner in the training the decay will saturate to
+                     its final value. beta=15 is ~40% of the training process.
+        """
+        # Create EMA
+        self.ema = deepcopy(kd_model)
+        self.ema.module.teacher = kd_model.module.teacher
+
+        # We don't freeze the teacher because we only have a shallow copy of it
+        self.ema.module.student.eval()
+
+        if exp_activation:
+            self.decay_function = lambda x: decay * (1 - math.exp(-x * beta))  # decay exponential ramp (to help early epochs)
+        else:
+            self.decay_function = lambda x: decay  # always return the same decay factor
+
+        """"
+        we hold a list of model attributes (not wights and biases) which we would like to include in each 
+        attribute update or exclude from each update. a SgModule declare these attribute using 
+        get_include_attributes and get_exclude_attributes functions. for a nn.Module which is not a SgModule
+        all non-private (not starting with '_') attributes will be updated (and only them).
+        """
+        self.include_attributes = kd_model.module.get_include_attributes()
+        self.exclude_attributes = kd_model.module.get_exclude_attributes()
+
+        # We don't freeze the teacher because we work with a reference of it and not a deepcopy, and we don't want to
+        # freeze the original
+        for p in self.ema.module.student.parameters():
+            p.requires_grad_(False)
+
+    def update(self, kd_model: KDModule, training_percent: float):
```

Comment:
See above comment about using super

Reply:
Changed in 35e859ad4b8293aa5c0992dfd3f2f1103a3f159d as well